go/arrow/array/numericbuilder.gen.go.tmpl

```diff
@@ -184,6 +184,124 @@ func (b *{{.Name}}Builder) newData() (data *Data) {
 	return
 }
 
+func (b *{{.Name}}Builder) AppendValueFromString(s string) error {
+	if s == NullValueStr {
+		b.AppendNull()
+		return nil
+	}
+  {{if or (eq .Name "Date32") -}}
+  	tm, err := time.Parse("2006-01-02", s)
+    if err != nil {
+      b.AppendNull()
+      return err
+    }
+    b.Append(arrow.Date32FromTime(tm))
```

Comment:
fix the indentation formatting please

Reply:
The problem with the indentation here is because in this file we use tabs instead of spaces. should we all turn it into spaces or should I keep using tabs in this file while in all `.go` files we do use spaces I believe. wdyt?

Reply:
I'm fine with converting the entire file to spaces. Go for it